pkg/api/server/server.go

```diff
@@ -82,78 +85,81 @@ func (s *Server) wrap(f api.HandlerFunc) http.HandlerFunc {
 			return
 		}
 
-		if strings.HasPrefix(req.URL.Path, "/api/") {
-			// Setup a new response writer for audit logging.
-			rw = &responseWriter{
-				ResponseWriter: rw,
-				auditEntry: audit.LogEntry{
-					Time:      time.Now(),
-					UserID:    user.GetUID(),
-					Method:    req.Method,
-					Path:      req.URL.Path,
-					UserAgent: req.UserAgent(),
-					SourceIP:  getSourceIP(req),
-					Host:      req.Host,
-				},
-				auditLogger: s.auditLogger,
-			}
+		// Apply per-user rate limits to all endpoints
+		s.rateLimiter.Limit(user, rw, req, func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
I'm not thrilled with how this method signature turned out, especially with the nested anonymous functions.

Besides that, this is actually creating a new anonymous function for _every_ request made to the server. I'm not sure how badly that will affect performance.

Anyone have an alternative approach in mind for this?

Reply:
Came up with something slightly less bad.